### PR TITLE
[Post Re-Invent]: Support for JSON with comments for security scans

### PR DESCRIPTION
**Pull request by laileni-aws**
## Problem
Json with comments `jsonc` is not supported for security scans.
## Solution
- Enabling support for `jsonc` for Codewhisperer security scans

- Telemetry languageid  will be same as json: `json`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by justinmk3 — APPROVED** [review]
Waiting for cw team approval

File: src/codewhisperer/util/runtimeLanguageContext.ts
Comment (andrewyuq) [inline src/codewhisperer/util/runtimeLanguageContext.ts:100]: can you check if this works with STE passing json(or it's variant) as the payload and didn't throw a validationException?

Comment (laileni-aws) [reply src/codewhisperer/util/runtimeLanguageContext.ts:100]: You mean to write a test case to validate the json or jsonc support to security scans?

Comment (andrewyuq) [reply src/codewhisperer/util/runtimeLanguageContext.ts:100]: not only for getDependencyGraph, but also the SendTelemetryEvent with codeScanEvent as the payload.

File: src/test/codewhisperer/service/codewhisperer.test.ts
Comment (justinmk3) [inline src/test/codewhisperer/service/codewhisperer.test.ts:79]: do we seriously need 10 lines of test code for each new filetype?

Comment (laileni-aws) [reply src/test/codewhisperer/service/codewhisperer.test.ts:79]: > not only for getDependencyGraph, but also the SendTelemetryEvent with codeScanEvent as the payload.

@andrewyuq requested to add test case to send json as payload to sendTelemetryEvent.
If thats ok I can drop this test case ?

Comment (andrewyuq) [reply src/test/codewhisperer/service/codewhisperer.test.ts:79]: The test I am referring to is not related to "telemetry optout status". We don't need to test "telemetry optout status" for different languages. They are independent fields
I was asking for a test that SendTelemetryEvent with codeScanEvent as the payload for the json language, that the request can succeed without any validation error.

Comment (laileni-aws) [reply src/test/codewhisperer/service/codewhisperer.test.ts]: > I was asking for a test that SendTelemetryEvent with codeScanEvent as the payload for the json language, that the request can succeed without any validation error.

If I understand correctly there is only 1 telemetry event for security scans(`codewhisperer_securityScan`) Previously we tested this event by passing python params. So, correct me if you something like [this](https://github.com/aws/aws-toolkit-vscode/blob/master/src/test/codewhisperer/commands/startSecurityScan.test.ts) to test for json too?

```
        assertTelemetry('codewhisperer_securityScan', {
            codewhispererLanguage: 'json',
            codewhispererCodeScanTotalIssues: 1,
            codewhispererCodeScanIssuesWithFixes: 0,
            codewhispererCodeScanLines: 188,
        } as CodewhispererSecurityScan)
```

Comment (andrewyuq) [reply src/test/codewhisperer/service/codewhisperer.test.ts]: A manual test is actually fine.

Comment (laileni-aws) [reply src/test/codewhisperer/service/codewhisperer.test.ts]: Tested manually and this works fine.

**Review by justinmk3 — APPROVED** [review]

**Review by andrewyuq — APPROVED** [review]

**Pull request merged**
